primitives/state-machine/src/ext.rs

```diff
@@ -460,20 +472,23 @@ where
 				target: "trie",
 				"Refuse to directly clear prefix that is part or contains of child storage key",
 			);
-			return (false, 0)
+			return (None, 0, 0, 0)
 		}
 
 		self.mark_dirty();
-		self.overlay.clear_prefix(prefix);
-		self.limit_remove_from_backend(None, Some(prefix), limit)
+		let overlay_count = self.overlay.clear_prefix(prefix);
+		let (next_cursor, backend_count, iterations) =
+			self.limit_remove_from_backend(None, Some(prefix), maybe_limit, maybe_cursor);
+		(next_cursor, backend_count, overlay_count + backend_count, iterations)
 	}
 
 	fn clear_child_prefix(
 		&mut self,
 		child_info: &ChildInfo,
 		prefix: &[u8],
-		limit: Option<u32>,
-	) -> (bool, u32) {
+		maybe_limit: Option<u32>,
+		maybe_cursor: Option<&[u8]>,
+	) -> (Option<Vec<u8>>, u32, u32, u32) {
```

Comment:
It *could*, except that it would require moving that into this crate and making it work with host function I/O which is more code than I want to write right now.

Reply:
Not when the API is currently still in V2? I think the intention is to only introduce `ClearPrefixResult` in V3.